lib/linux_admin/ssh.rb

```diff
@@ -0,0 +1,94 @@
+module LinuxAdmin
+  class SSH
+    attr_reader :ip
+    attr_reader :username
+    attr_reader :private_key
+    attr_reader :agent
+
+    def initialize(ip, username, private_key)
+      @ip = ip
+      @private_key = private_key
+      @username = username
+    end
+
+    def perform_commands(commands = [], options = {})
+      require 'net/ssh'
+      if options[:agent_enabled]
```

Comment:
This could also be improved if the socket was passed in the initializer. That way we don't have to reference the SSHAgent class from here at all.